src/DynamoCore/Graph/ModelBase.cs

```diff
@@ -272,23 +272,26 @@ public bool UpdateValue(UpdateValueParams updateValueParams)
         }
 
         /// <summary>
-        /// This method is currently used as a way to send an event to ModelBase 
-        /// derived objects. Its primary use is in DynamoNodeButton class, which 
-        /// sends this event when clicked.
+        /// This method is currently used as a way to send an event to ModelBase
+        /// derived objects. Its primary use is in DynamoNodeButton class, which
+        /// sends this event when clicked, to change the number of ports in a
+        /// VariableInputNode.
         /// </summary>
         /// <param name="eventName">The name of the event.</param>
+        /// <param name="value">For the SetInPortCount event, the number of
+        /// ports desired.  Ignored for other events.</param>
         /// <param name="recorder"></param>
         /// <returns>Returns true if the call has been handled, or false otherwise.
         /// </returns>
-        internal bool HandleModelEvent(string eventName, UndoRedoRecorder recorder)
+        internal bool HandleModelEvent(string eventName, int value, UndoRedoRecorder recorder)
```

Comment:
So we've added `int value` here solely to support the case of changing the number of ports? All other calls to this will need to have this value defined?